Lib/zipfile/__init__.py

```diff
@@ -331,7 +331,7 @@ def _EndRecData(fpin):
     # number does not appear in the comment.
     maxCommentStart = max(filesize - (1 << 16) - sizeEndCentDir, 0)
```

Comment:
While not part of your changes, I think this line might be wrong. It should allow up to `ZIP_MAX_COMMENT = (1 << 16) - 1` bytes for the comment (one less than currently)

```suggestion
    maxCommentStart = max(filesize - ZIP_MAX_COMMENT - sizeEndCentDir, 0)
```

Reply:
Was wondering about that. Does it makes sense to make a variable for `ZIP_MAX_COMMENT + sizeEndCentDir` variable for both spots to use? (if the `fpin.seek` jus below can be changed to a SEEK_END like the previous seek, this code also no longer needs to do its own `filesize` fetching + calculation)

Reply:
You could make a variable for `ZIP_MAX_COMMENT + sizeEndCentDir` but I'm happy either way.

With regards to the `filesize` variable, you could rewrite it to avoid that first `fpin.seek(0, 2)` but we still need to have the absolute location of the start of the central directory as we use it for this line in `_RealGetContents`:
`        self.start_dir = offset_cd + concat`
So you could get the (potential) central directory offset directly using a `tell()` after
```
    try:
        fpin.seek(-sizeEndCentDir, 2)
    except OSError:
        return None
```
succeeds and do the rest of the offset calculations relative to that. That would be one less `seek()`. Probably best to do that in a different PR if you want

Reply:
Will leave as is for now / focused on other bits around I/O stack